ci/buildspec_copy_image.yml

```diff
@@ -0,0 +1,53 @@
+# AWS CodeBuild config
+# This file is referenced in the infra repo as a CodeBuild buildspec target.
+
+## The environment variables in this file are passed in through the buildbuild resource.
+## Check the terraform scripts to see where these variables originate from.
+
+version: 0.2
+env:
+  shell: bash
+phases:
+  build:
+    commands:
+      # Use SRC_REPO_ACCOUNT_ID if set, otherwise default to ECR_ACCOUNT_ID
+      - export SRC_REPO_ACCOUNT_ID="${SRC_REPO_ACCOUNT_ID:-${ECR_ACCOUNT_ID}}"
+      # Use SRC_REPO_REGION if set, otherwise default to ECR_REGION
+      - export SRC_REPO_REGION="${SRC_REPO_REGION:-${ECR_REGION}}"
+      # Use DEST_REPO_ACCOUNT_ID if set, otherwise default to ECR_ACCOUNT_ID
+      - export DEST_REPO_ACCOUNT_ID="${DEST_REPO_ACCOUNT_ID:-${ECR_ACCOUNT_ID}}"
+      # Use DEST_REPO_REGION if set, otherwise default to ECR_REGION
+      - export DEST_REPO_REGION="${DEST_REPO_REGION:-${ECR_REGION}}"
+
+      # Build host values from repo vars
+      - export SRC_REPO_HOST="${SRC_REPO_ACCOUNT_ID}.dkr.ecr.${SRC_REPO_REGION}.amazonaws.com"
+      - export DEST_REPO_HOST="${DEST_REPO_ACCOUNT_ID}.dkr.ecr.${DEST_REPO_REGION}.amazonaws.com"
+
+      # Log in to source
+      - aws ecr get-login-password --region "${SRC_REPO_REGION}" | docker login --username AWS --password-stdin "${SRC_REPO_HOST}"
+
+      # If destination is different, log into that, too
+      - > 
+        if [[ "${SRC_REPO_HOST}" == "${DEST_REPO_HOST}" ]]; then 
+          echo "Source and destination are the same"
+        else
+          aws ecr get-login-password --region "${DEST_REPO_REGION}" | docker login --username AWS --password-stdin "${DEST_REPO_HOST}"
+        fi
+
+      - export SOURCE_REPO="${SRC_REPO_HOST}/${SOURCE_REPO_NAMESPACE}"
+      - export SOURCE_IMAGE_URL="${SOURCE_REPO}/${SOURCE_IMAGE}"
```

Comment:
Curious- are we specifying an image with a SHA tag? If not, we may be introducing a race condition.

Reply:
It's configurable, but most likely not.  This is used as a bridge between environments.  By the time we copy the backend image over from dev to staging, for example, all tests have passed and it has successfully been deployed into dev.  Even if a new change is pending in the pipeline (which won't run until the current pipeline execution is finished), the image we copy over is known good.

Reply:
We'd do something like `doorway-dev/backend:run` -> `doorway-staging/backend:run-candidate` and then handle the deployment from there more or less how we do already.  This is just a fast and cost-effective way to move the image through the pipeline without having to rebuild